common/djangoapps/student/tests/test_models.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-docstring
+
+import hashlib
+
+from django.contrib.auth.models import AnonymousUser
+from django.core.cache import cache
+from django.db.models.functions import Lower
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory, CourseEnrollmentFactory
+
+
+class CourseEnrollmentTests(SharedModuleStoreTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(CourseEnrollmentTests, cls).setUpClass()
+        cls.course = CourseFactory()
+
+    def setUp(self):
+        super(CourseEnrollmentTests, self).setUp()
+        self.user = UserFactory.create()
+
+    def test_enrollment_status_hash_cache_key(self):
+        username = 'test-user'
+        user = UserFactory(username=username)
+        expected = 'enrollment_status_hash_' + username
+        self.assertEqual(CourseEnrollment.enrollment_status_hash_cache_key(user), expected)
+
+    def assert_enrollment_status_hash_cached(self, user, expected_value):
+        self.assertEqual(cache.get(CourseEnrollment.enrollment_status_hash_cache_key(user)), expected_value)
```

Comment:
Since your tests here hit the cache, you should make this test class inherit from `CacheIsolationTestCase` and you should set `ENABLED_CACHES = ['default']`. Here's [an example](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/programs/tasks/v1/tests/test_tasks.py#L57-L61). This class helps with parallelized execution of tests and also makes sure the cache is cleaned out between test runs.